### ISSUE TITLE
Don't use from_utf8_unchecked on network data

### ISSUE THREAD

**Issue by CodesInChaos** [rust-gjk/ChatApp#1]
Data from received from the network is untrusted, but rust relies on strings being valid UTF-8 for safety. Replace `from_utf8_unchecked` with `from_utf8` to avoid this potential security hole.

**Comment by luciusmagn** [comment]
You are absolutely correct. I did that as a shortcut (because we are sending data from Rust to Rust) and as a temporary solution. We will soon be sending proper packets instead of plain text
